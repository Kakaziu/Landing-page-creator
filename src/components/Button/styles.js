import styled, { css } from 'styled-components';

const buttonSizes = {
  small: (theme) => css`
    padding: 8px 20px;
    font-size: ${theme.fonts.sizes.small};
  `,

  big: (theme) => css`
    padding: 8px 30px;
    font-size: ${theme.fonts.sizes.medium};
  `,
};

export const LinkContainer = styled.a``;

export const Button = styled.button`
  ${({ theme, size }) => css`
    font-weight: bold;
    border: none;
    background-color: ${theme.colors.secondaryColor};
    color: ${theme.colors.white};
    border-radius: 4px;
    ${buttonSizes[size](theme)}
  `}
`;
