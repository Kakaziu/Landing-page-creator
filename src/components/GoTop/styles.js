import styled, { css } from 'styled-components';

export const Button = styled.a`
  ${({ theme }) => css`
    position: fixed;
    bottom: 4rem;
    right: 4rem;
    width: 40px;
    height: 40px;
    border-radius: 20px;
    border: none;
    display: flex;
    align-items: center;
    justify-content: center;
    background-color: rgba(10, 18, 40, 0.5);
    transition: all 0.3s ease-in-out;
    cursor: pointer;

    &:hover {
      background-color: ${theme.colors.primaryColor};
    }
  `}
`;
