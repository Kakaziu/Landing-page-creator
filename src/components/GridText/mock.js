export default {
  background: false,
  title: 'My grid',
  description:
    'Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde',
  gridText: [
    {
      title: 'Teste 1',
      description:
        'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.',
    },
    {
      title: 'Teste 2',
      description:
        'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.',
    },
    {
      title: 'Teste 3',
      description:
        'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.',
    },
  ],
  button: {
    textButton: 'Clique aqui',
    urlButton: '#',
    size: 'small',
  },
};
