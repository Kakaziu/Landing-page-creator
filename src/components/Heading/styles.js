import styled, { css } from 'styled-components';

const titleSizes = {
  small: (theme) => css`
    font-size: ${theme.fonts.sizes.medium};
  `,

  medium: (theme) => css`
    font-size: ${theme.fonts.sizes.large};
  `,

  big: (theme) => css`
    font-size: ${theme.fonts.sizes.xxlarge};
  `,

  huge: (theme) => css`
    font-size: ${theme.fonts.sizes.xhuge};
    ${mediaFont(theme)}
  `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.fonts.sizes.xlarge};
  }
`;

export const Title = styled.h1`
  ${({ theme, isdark, uppercase, size }) => css`
    color: ${isdark ? theme.colors.primaryColor : theme.colors.white};
    text-transform: ${uppercase ? 'uppercase' : 'none'};
    ${titleSizes[size](theme)}
  `}
`;
