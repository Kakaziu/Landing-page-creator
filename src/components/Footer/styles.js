import styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    align-items: center;
    border-top: 1px solid ${theme.colors.mediumGray};
    font-size: ${theme.fonts.sizes.small};
    padding: ${theme.spacings.small};

    a {
      color: ${theme.colors.primaryColor};
      text-decoration: none;
      font-size: ${theme.fonts.sizes.small};
    }

    & ${SectionContainer} {
      padding: ${theme.spacings.small} 0;
    }
  `}
`;
