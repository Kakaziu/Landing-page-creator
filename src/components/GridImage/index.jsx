import P from 'prop-types';
import SectionBackground from '../SectionBackground';
import * as Styled from './styles';
import Heading from '../Heading';
import Button from '../Button';
import TextComponent from '../TextComponent';

const GridImage = ({
  title,
  description,
  grid = [],
  background = false,
  button,
}) => {
  return (
    <SectionBackground background={background}>
      <Styled.Container background={+background}>
        <Styled.TitleContainer>
          <Heading uppercase>{title}</Heading>
          <TextComponent>{description}</TextComponent>
        </Styled.TitleContainer>
        <Styled.GridContainer>
          {grid.map((element) => (
            <Styled.GridElement key={element.altText}>
              <img src={element.srcImg} alt={element.altText} />
            </Styled.GridElement>
          ))}
        </Styled.GridContainer>
        {button.children && (
          <Button url={button.urlButton} size={button.size}>
            {button.children}
          </Button>
        )}
      </Styled.Container>
    </SectionBackground>
  );
};

GridImage.propTypes = {
  title: P.string.isRequired,
  description: P.string.isRequired,
  grid: P.arrayOf(
    P.shape({
      srcImg: P.string.isRequired,
      altText: P.string.isRequired,
    }),
  ),
  background: P.bool,
  button: Button.propTypes,
};

export default GridImage;
