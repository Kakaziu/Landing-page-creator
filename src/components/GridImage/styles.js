import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { LinkContainer as Button } from '../Button/styles';

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      margin-bottom: ${theme.spacings.small};
    }

    img {
      width: 100%;
      transition: all 300ms ease-in-out;
      cursor: pointer;

      &:hover {
        transform: scale(1.1);
      }
    }
  `}
`;

export const TitleContainer = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      margin-bottom: ${theme.spacings.small};
    }
  `}
`;

export const GridContainer = styled.div`
  ${({ theme }) => css`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: ${theme.spacings.xlarge};
  margin-top: ${theme.spacings.huge};
  width: 100%;

  @media ${theme.media.lteMedium} {
    grid-template-columns: 1fr;
  }
`}
`;

export const Container = styled.div`
  ${({ theme, background }) => css`
    margin: 8rem 0;
    ${HeadingContainer} {
      margin-bottom: ${theme.spacings.small};
      color: ${
        background === 1 ? theme.colors.white : theme.colors.primaryColor
      }
    }

    ${Button} {
      margin-top: ${theme.spacings.large};
      text-align: center;
    }
  `}
`;
