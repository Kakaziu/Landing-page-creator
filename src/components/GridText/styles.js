import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      margin-bottom: ${theme.spacings.small};
    }
  `}
`;

export const TitleContainer = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      margin-bottom: ${theme.spacings.small};
    }
  `}
`;

export const GridContainer = styled.div`
  ${({ theme }) => css`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: ${theme.spacings.xlarge};
  margin-top: ${theme.spacings.huge};
  width: 100%;

  @media ${theme.media.lteMedium} {
    grid-template-columns: 1fr;
  }
`}
`;

export const Container = styled.div`
  ${({ theme, background }) => css`
    ${HeadingContainer} {
      margin-bottom: ${theme.spacings.small};
      color: ${
        background === 1 ? theme.colors.white : theme.colors.primaryColor
      }
    }
  `}
`;
