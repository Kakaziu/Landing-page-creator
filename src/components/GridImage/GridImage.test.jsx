import { screen } from '@testing-library/react';
import GridImage from '.';
import renderTheme from '../../styles/render-theme';
import { theme } from '../../styles/theme';
import mock from './mock';

describe('<GridImage/>', () => {
  it('should render with background', () => {
    const { container } = renderTheme(<GridImage {...mock} background />);
    expect(container).toMatchSnapshot();
    expect(container.firstChild).toHaveStyle({
      background: theme.colors.primaryColor,
    });
  });

  it('should render without background', () => {
    const { container } = renderTheme(<GridImage {...mock} />);

    expect(container.firstChild).toHaveStyle({
      background: theme.colors.white,
    });
  });

  it('should not render grid when no have grid', () => {
    renderTheme(<GridImage />);

    expect(screen.queryAllByAltText(/teste/i)).toHaveLength(0);
  });
});
