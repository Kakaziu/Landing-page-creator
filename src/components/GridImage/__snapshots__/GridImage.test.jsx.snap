// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridImage/> should render with background 1`] = `
.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c0 {
  background-color: #0A1128;
  color: #FFFFFF;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c5 {
  color: #0A1128;
  text-transform: uppercase;
  font-size: 3.2rem;
}

.c10 {
  display: block;
}

.c11 {
  font-weight: bold;
  border: none;
  background-color: #dc143c;
  color: #FFFFFF;
  border-radius: 4px;
  padding: 8px 40px;
  font-size: 2.4rem;
  cursor: pointer;
  transition: all 0.3s ease-in-out;
}

.c11:hover {
  background-color: #b81132;
}

.c8 .c4 {
  margin-bottom: 1.6rem;
}

.c8 img {
  width: 100%;
  transition: all 300ms ease-in-out;
  cursor: pointer;
}

.c8 img:hover {
  transform: scale(1.1);
}

.c3 .c4 {
  margin-bottom: 1.6rem;
}

.c7 {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: 4.0rem;
  margin-top: 5.6rem;
  width: 100%;
}

.c2 .c4 {
  margin-bottom: 1.6rem;
  color: #FFFFFF;
}

.c2 .c9 {
  margin-top: 3.2rem;
  text-align: center;
}

.c6 {
  font-size: 1.6rem;
}

@media (max-width:768px) {
  .c7 {
    grid-template-columns: 1fr;
  }
}

<div>
  <div
    background="1"
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        background="1"
        class="c2"
      >
        <div
          class="c3"
        >
          <h1
            class="c4 c5"
            isdark="1"
            uppercase="1"
          >
            My grid
          </h1>
          <p
            class="c6"
          >
            Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
          </p>
        </div>
        <div
          class="c7"
        >
          <div
            class="c8"
          >
            <img
              alt="Teste 1"
              src="https://source.unsplash.com/random/800x800?r=1"
            />
          </div>
          <div
            class="c8"
          >
            <img
              alt="Teste 2"
              src="https://source.unsplash.com/random/800x800?r=2"
            />
          </div>
          <div
            class="c8"
          >
            <img
              alt="Teste 3"
              src="https://source.unsplash.com/random/800x800?r=3"
            />
          </div>
          <div
            class="c8"
          >
            <img
              alt="Teste 4"
              src="https://source.unsplash.com/random/800x800?r=4"
            />
          </div>
          <div
            class="c8"
          >
            <img
              alt="Teste 5"
              src="https://source.unsplash.com/random/800x800?r=5"
            />
          </div>
          <div
            class="c8"
          >
            <img
              alt="Teste 6"
              src="https://source.unsplash.com/random/800x800?r=6"
            />
          </div>
        </div>
        <a
          class="c9 c10"
          href="#"
        >
          <button
            class="c11"
          >
            Clique aqui
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
